pkg/localstorev2/internal/reserve/items.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package reserve
+
+import (
+	"encoding/binary"
+
+	storage "github.com/ethersphere/bee/pkg/storagev2"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+type batchRadiusItem struct {
+	Bin     uint8
+	BatchID []byte
+	Address swarm.Address
+	BinID   uint64
+}
+
+func (b *batchRadiusItem) Namespace() string {
+	return "batchRadius"
+}
+
+// bin/batchID/ChunkAddr
+func (b *batchRadiusItem) ID() string {
+	return batchBinToString(b.Bin, b.BatchID) + b.Address.ByteString()
+}
+
+func batchBinToString(bin uint8, batchID []byte) string {
+	return string(bin) + string(batchID)
+}
+
+func (b *batchRadiusItem) Clone() storage.Item {
+	return &batchRadiusItem{
+		Bin:     b.Bin,
+		BatchID: b.BatchID,
+		Address: b.Address,
+		BinID:   b.BinID,
+	}
+}
+
+func (b *batchRadiusItem) String() string {
+	return ""
+}
+
+const batchRadiusItemSize = 1 + swarm.HashSize + swarm.HashSize + 8
+
+func (b *batchRadiusItem) Marshal() ([]byte, error) {
+
+	buf := make([]byte, batchRadiusItemSize)
+
+	i := 0
```

Comment:
So I dont understand how this is more error prone?

The way you are copying is actually error-prone as you are giving access to the whole buffer
`buf[i:]` instead of the exact buffer which is preallocated `buf[start:end]`. Also knowing the start and end indexes from the code is better than following value of `i`.

Also, this makes it inconsistent with the other packages as well. Not sure if there is a strong enough reason for this. Different ways introduce different problems.